### PR TITLE
make batch upgrade run parallel and set pvc selector

### PR DESCRIPTION
**Pull request by zwwhdls**
1. batch upgrade run in parallel
2. set parallel num in dashboard
3. set pvc selector in batch upgrade

CleanShot 2024-12-05 at 18 00 00

### CODE REVIEW

File: pkg/dashboard/batch.go [hunk 1]
```diff
@@ -329,7 +342,111 @@ func newUpgradeJob(nodeName string, recreate bool) *batchv1.Job {
...
+	s, _ = metav1.LabelSelectorAsSelector(ls)
+	if err = api.cachedReader.List(ctx, &secretList, &client.ListOptions{LabelSelector: s}); err != nil {
+		return nil, nil, err
+	}
+
+	pvMap := make(map[string]*corev1.PersistentVolume)
+	pvcMap := make(map[string]*corev1.PersistentVolumeClaim)
+	secretMap := make(map[string]*corev1.Secret)
+	for _, pv := range pvs {
+		pv2 := pv
```
Comment (zxh326) [inline pkg/dashboard/batch.go]: We've upgraded to 1.22, so we don't have to do this.

File: cmd/dashboard/upgrade.go [hunk 2]
```diff
@@ -18,98 +18,174 @@ package main
...
+					resultCh <- u.triggerUpgrade(ctx, &mp2)
+				}()
+			}
+		}()
+
+		for oneErr := range resultCh {
+			if oneErr != nil {
+				if !u.conf.IgnoreError {
+					fmt.Printf("%s BATCH-FAIL some pods upgrade failed\n", time.Now().Format(time.DateTime))
+					return
```
Comment (zxh326) [inline cmd/dashboard/upgrade.go:153]: What happens to the other pods that are being upgraded after the return here?

File: go.mod [hunk 3]
```diff
@@ -16,6 +16,8 @@ require (
 	github.com/smartystreets/goconvey v1.6.4
 	github.com/spf13/cobra v1.8.1
 	github.com/stretchr/testify v1.9.0
+	github.com/sykesm/zap-logfmt v0.0.4
+	go.uber.org/zap v1.26.0
```
Comment (zxh326) [inline go.mod]: not used

**Review by zxh326 — COMMENTED** [review]
run `pnpm run format` in the dashboard-ui

File: cmd/dashboard/upgrade.go (hunk 2 above)
Comment (zwwhdls) [reply cmd/dashboard/upgrade.go:153]: complete upgrading until fail or success. We can not cancel it.

**Review by zxh326 — APPROVED** [review]

**Pull request merged**